pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarRegistrationClient.java

```diff
@@ -153,42 +178,57 @@ public void unwatchReadOnlyBookies(RegistrationListener registrationListener) {
         readOnlyBookiesWatchers.remove(registrationListener);
     }
 
-    private void handleDeletedBookieNode(Notification n) {
-        if (n.getType() == NotificationType.Deleted) {
-            BookieId bookieId = stripBookieIdFromPath(n.getPath());
-            if (bookieId != null) {
-                log.info("Bookie {} disappeared", bookieId);
-                bookieServiceInfoCache.remove(bookieId);
-            }
+    /**
+     * This method will receive metadata store notifications and then update the
+     * local cache in background sequentially.
+     */
+    private void updatedBookies(Notification n) {
+        // make the notification callback run sequential in background.
+        final String path = n.getPath();
+        if (!path.startsWith(bookieReadonlyRegistrationPath) || !path.startsWith(bookieRegistrationPath)) {
```

Comment:
It should be `&&` here? Otherwise, we will miss the update from the `bookieRegistrationPath` because `!path.startsWith(bookieReadonlyRegistrationPath)` will be true.

Reply:
fixed